coderd/coderd.go

```diff
@@ -611,6 +620,20 @@ func New(options *Options) *API {
 					r.Use(httpmw.ExtractOAuth2(options.OIDCConfig, options.HTTPClient, oidcAuthURLParams))
 					r.Get("/", api.userOIDC)
 				})
+				// This is an authenticated route. It handles converting a user
+				// from Password based authentication to Oauth
+				r.Route("/convert-login", func(r chi.Router) {
+					r.Use(
+						apiKeyMiddleware,
+					)
+					if allowOauthConversion {
+						r.Post("/", api.postConvertLoginType)
+					} else {
+						r.Post("/", func(rw http.ResponseWriter, r *http.Request) {
+							http.Error(rw, "Oauth conversion is not allowed, contact an administrator to turn on this feature.", http.StatusForbidden)
+						})
+					}
```

Comment:
It's weird to not include this in the route IMO

Reply:
I didn't want to trigger an audit log. I'll just throw it at the top of the route func